### ISSUE TITLE
Extremely strange behavior of ReactiveForms inside Dialog

### ISSUE THREAD

**Issue by montella1507** [angular/components#28165]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Please check the minimal reproduction here:
https://stackblitz.com/edit/components-issue-tkdwp5?file=src%2Fapp%2Fexample-component.ts,src%2Fapp%2Fexample-component.html

1) Custom "form control" component implementing Ng_Async_validator and CVA = CustomComponent. Inside, there is innerForm with 2 controls - A and B. There is AsyncValidator on entire group.
2) This custom control "Custom" is used inside "InnerComponent" 
`<app-custom [formControl]="form.controls.custom"></app-custom>`
3) If it is used inside Dialog.  Those 2 controls inside Custom component will get Errors from parent component, but this is bug. 
<img width="470" alt="image" src="https://github.com/angular/components/assets/10418323/a59c5678-d353-4d56-9ac4-eb04988f3c06">

+ errors and state IS NOT correctly propagated via validate() interface to parent.
+ validate() function of AsyncValidator interface is called 2 times when value is changed (but should have been called only ONCE!)

-------
IF YOU DONT use dialog => change the code to this
<img width="675" alt="image" src="https://github.com/angular/components/assets/10418323/dad1e6c5-dc9f-42f5-8ff0-e3e84c5a9c58">

It works LIKE A CHARM.

It looks lika Dialog is somehow destroying "providers" tree and it somehow change the way how is the CVA / AsyncValidator resolved for ... everything inside..?

### Reproduction

https://stackblitz.com/edit/components-issue-tkdwp5?file=src%2Fapp%2Fexample-component.ts,src%2Fapp%2Fexample-component.html

### Expected Behavior

Dialog should have not changed the behavior of CVA / AsyncValidator implementation

Here is the video how it behaves (inside dialog) vs. how it should have behaved:

https://github.com/angular/components/assets/10418323/5bd09e0a-5999-4a04-92be-ca62182e99c0

I have been debuging this issue for 4 hours now...

### Actual Behavior

Dialog change the behavior of things inside.

### Environment

"dependencies": {
    "@angular/animations":"15.1.4",
    "@angular/cdk": "15.1.4",
    "@angular/common": "15.1.4",
    "@angular/compiler": "15.1.4",
    "@angular/core":"15.1.4",
    "@angular/forms":"15.1.4",
    
    "@angular/platform-browser": "15.1.4",
    "@angular/platform-browser-dynamic":"15.1.4",
    "@angular/router": "15.1.4",
    "angular-in-memory-web-api": "^0.17.0",
    "core-js": "^3.6.5",
    "hammerjs": "^2.0.8",
    "moment": "^2.29.4",
    "rxjs": "^7.8.1",
    "tslib": "^2.6.2",
    "zone.js": "0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "15.1.5",
    "@angular/cli": "~15.1.0",
    "@angular/compiler-cli": "15.1.4",
    "@angular/language-service": "^9.0.0",
    "@angular/localize": "^9.0.0",
    "@types/node": "^12.12.6",
    "codelyzer": "^5.2.0",
    "ts-node": "^8.4.1",
    "tslint": "^5.20.1",
    "typescript": "4.8.4"
  }

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
